core/creator/src/test/java/io/quarkus/creator/phase/runnerjar/test/VariableSubstitutionTest.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.creator.phase.runnerjar.test;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+/**
+ * This tests the core logic of the variable substitution;
+ */
+public class VariableSubstitutionTest {
+    private static final String PLACEHOLDER_PREFIX = "${";
+    private static final String PLACEHOLDER_SUFFIX = "}";
+
+    @Test
+    public void testVariableSubstitution() {
```

Comment:
This is not really a test, but we don't really have any test infrastructure at the moment for running the generated jar files (they get tested indirectly via the native image tests). You could probably write a real test in integration/main  using @NativeImageTest